dipy/workflows/denoise.py

```diff
@@ -96,19 +107,33 @@ def run(
                 logging.info("Denoising %s", fpath)
                 data, affine, image = load_nifti(fpath, return_img=True)
                 bvals = np.loadtxt(bvalpath)
-
-                denoised_data = patch2self(
-                    data,
-                    bvals,
-                    model=model,
-                    b0_threshold=b0_threshold,
-                    alpha=alpha,
-                    verbose=verbose,
-                    patch_radius=patch_radius,
-                    b0_denoising=b0_denoising,
-                    clip_negative_vals=clip_negative_vals,
-                    shift_intensity=shift_intensity,
-                )
+                if ver == 1:
+                    denoised_data = patch2self(
+                        data,
+                        bvals,
+                        model=model,
+                        b0_threshold=b0_threshold,
+                        alpha=alpha,
+                        verbose=verbose,
+                        patch_radius=patch_radius,
+                        b0_denoising=b0_denoising,
+                        clip_negative_vals=clip_negative_vals,
+                        shift_intensity=shift_intensity,
+                        version=ver,
+                    )
+                elif ver == 3:
+                    denoised_data = patch2self(
+                        data,
+                        bvals,
+                        model=model,
+                        b0_threshold=b0_threshold,
+                        alpha=alpha,
+                        verbose=verbose,
+                        b0_denoising=b0_denoising,
+                        clip_negative_vals=clip_negative_vals,
+                        shift_intensity=shift_intensity,
+                        version=ver,
+                    )
```

Comment:
Why do you need a `if` here ? you call the exact same function with the same parameters....

Reply:
Thank you for the review @skoudoro. In this case, we need the if condition as patch_radius argument is not supported in patch2self3. It only works in patch2self version 1.